inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/SchemaChangeUtils.java

```diff
@@ -52,49 +54,85 @@ public class SchemaChangeUtils {
     static List<TableChange> diffSchema(Schema oldSchema, Schema newSchema) {
         List<String> oldFields = oldSchema.columns().stream().map(NestedField::name).collect(Collectors.toList());
         List<String> newFields = newSchema.columns().stream().map(NestedField::name).collect(Collectors.toList());
-        int oi = 0;
-        int ni = 0;
+        Set<String> oldFieldSet = new HashSet<>(oldFields);
+        Set<String> newFieldSet = new HashSet<>(newFields);
+
+        Set<String> intersectColSet = Sets.intersection(oldFieldSet, newFieldSet);
+        Set<String> colsToDelete = Sets.difference(oldFieldSet, newFieldSet);
+        Set<String> colsToAdd = Sets.difference(newFieldSet, oldFieldSet);
+
         List<TableChange> tableChanges = new ArrayList<>();
-        while (ni < newFields.size()) {
-            if (oi < oldFields.size() && oldFields.get(oi).equals(newFields.get(ni))) {
-                oi++;
-                ni++;
-            } else {
-                NestedField newField = newSchema.findField(newFields.get(ni));
+
+        // step0: judge whether unknown change
+        // just diff two different schema can not distinguish（add + delete) vs modify
+        // Example first [a, b, c] -> then delete c [a, b] -> add d [a, b, d], currently it is only judged as unknown
+        // change.
+        // In next version,we will judge it is [delete and add] or rename by using information extracted from DDL
+        if (!colsToDelete.isEmpty() && !colsToAdd.isEmpty()) {
```

Comment:
whether to add a reorder column check?

Reply:
Excuse me, could you elaborate more?

Reply:
[a,b,c]->[a,c,b]
it'a possible to happen，so you should consider it as an abnormal case or just ignore it.It should clearly described in the method comment. And you should add some ut to validate the input and output.

Reply:
Thanks,I get it and I will perfect it

Reply:
I added judge field position change, please check again